### PR TITLE
Turn on a bunch more remote tests

### PR DESCRIPTION
**Pull request by rhatdan**
We need to be more specific about the remote tests we turn off.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
This PR will turn on another 100+ tests for podman --remote.

@edsantiago @vrothberg @giuseppe @mheon @baude @jwhonce @TomSweeneyRedHat @ashley-cui @QiWang19 PTAL

BTW, I think podman remote is not able to see images that are stored as localhost/image.

This is why the builds followed by runs are failing.

File: test/e2e/generate_systemd_test.go [hunk 1]
```diff
@@ -1,5 +1,3 @@
-// +build !remote
```
Comment (edsantiago) [inline test/e2e/generate_systemd_test.go:1]: Honestly, is there any point to enabling this one?

Comment (vrothberg) [reply test/e2e/generate_systemd_test.go:1]: `generate systemd` isn't supported for remote clients, so I think we can leave it fully disabled

File: test/e2e/run_entrypoint_test.go [hunk 2]
```diff
@@ -46,6 +44,7 @@ CMD []
 	})
 
 	It("podman run entrypoint", func() {
+		SkipIfRemote()
```
Comment (edsantiago) [inline test/e2e/run_entrypoint_test.go:47]: I'm not sure why this and the subsequent ones are skipped? They're all tagged as `foobar.com/something` so the `localhost` issue shouldn't apply. And I don't understand what you mean by `localhost` anyway, I'm playing with podman-remote on my system (with master) and am not having problems building, tagging, or running images with unqualified names...?

Comment (rhatdan) [reply test/e2e/run_entrypoint_test.go:47]: I just wanted to get as many in as possible.  I did not review why they were blowing up. I was hoping to assign each one to others who are closer to the issue, to investigate dropping the SkipIfRemote().

File: test/e2e/generate_systemd_test.go (hunk 1 above)
Comment (rhatdan) [reply test/e2e/generate_systemd_test.go]: Reverted.

**Comment by rhatdan** [comment]
Once we have this merged, we need to open a card for each test*go for someone to examine the skipremote tests to see if they could pass on remote systems.

**Comment by edsantiago** [comment]
I opened RUN-973 this morning for that, although it's not as fine-grain as your idea.

One suggestion: if more tests choke, could you add a one-liner comment next to `SkipIfRemote` indicating (very briefly) the nature of the failure? That could make it easier to find common failure causes without doing full runs. Thanks.

Given the understanding that these skips are temporary, this PR LGTM.

**Comment by edsantiago** [comment]
[CI error](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4561900547604480/html/integration_test.log.html) is a flake, #5212, and is exactly what I'm trying to address with #7047.

File: test/e2e/port_test.go [hunk 3]
```diff
@@ -127,18 +128,18 @@ var _ = Describe("Podman port", func() {
...
 		defer lock2.Unlock()
 
-		setup := podmanTest.Podman([]string{"run", "-dt", "-p", "5000:5000", "-p", "5001:5001", ALPINE, "top"})
+		setup := podmanTest.Podman([]string{"run", "--name", "test", "-dt", "-p", "5000:5000", "-p", "5001:5001", ALPINE, "top"})
 		setup.WaitWithDefaultTimeout()
 		Expect(setup.ExitCode()).To(BeZero())
 
 		// Check that the first port was honored
-		result1 := podmanTest.Podman([]string{"port", "-l", "5000"})
+		result1 := podmanTest.Podman([]string{"port", "test", "5000"})
```
Comment (TomSweeneyRedHat) [inline test/e2e/port_test.go:136]: should this still be `-l`?

File: test/e2e/run_networking_test.go [hunk 4]
```diff
@@ -265,10 +263,10 @@ var _ = Describe("Podman run networking", func() {
 	})
 
 	It("podman run network expose ports in image metadata", func() {
-		session := podmanTest.Podman([]string{"create", "-dt", "-P", nginx})
+		session := podmanTest.Podman([]string{"create", "--name", "test", "-dt", "-P", nginx})
 		session.Wait(90)
 		Expect(session.ExitCode()).To(Equal(0))
-		results := podmanTest.Podman([]string{"inspect", "-l"})
+		results := podmanTest.Podman([]string{"inspect", "test"})
```
Comment (TomSweeneyRedHat) [inline test/e2e/run_networking_test.go:269]: Should this really change?

File: test/e2e/run_restart_test.go [hunk 5]
```diff
@@ -35,11 +33,12 @@ var _ = Describe("Podman run restart containers", func() {
...
 
 	It("Podman start after successful run", func() {
-		session := podmanTest.Podman([]string{"run", ALPINE, "ls"})
+		SkipIfRemote()
+		session := podmanTest.Podman([]string{"run", "--name", "test", ALPINE, "ls"})
 		session.WaitWithDefaultTimeout()
 		Expect(session.ExitCode()).To(Equal(0))
 
-		session2 := podmanTest.Podman([]string{"start", "--attach", "--latest"})
+		session2 := podmanTest.Podman([]string{"start", "--attach", "test"})
```
Comment (TomSweeneyRedHat) [inline test/e2e/run_restart_test.go:41]: another latest question

File: test/e2e/port_test.go (hunk 3 above)
Comment (edsantiago) [reply test/e2e/port_test.go:136]: `-l` doesn't work with `remote`

File: test/e2e/top_test.go [hunk 6]
```diff
@@ -55,22 +53,22 @@ var _ = Describe("Podman top", func() {
 	})
 
 	It("podman top on container", func() {
-		session := podmanTest.Podman([]string{"run", "-d", ALPINE, "top", "-d", "2"})
+		session := podmanTest.Podman([]string{"run", "--name", "test", "-d", ALPINE, "top", "-d", "2"})
 		session.WaitWithDefaultTimeout()
 		Expect(session.ExitCode()).To(Equal(0))
 
-		result := podmanTest.Podman([]string{"top", "-l"})
+		result := podmanTest.Podman([]string{"top", "test"})
```
Comment (TomSweeneyRedHat) [inline test/e2e/top_test.go:60]: again latest

File: test/e2e/port_test.go (hunk 3 above)
Comment (TomSweeneyRedHat) [reply test/e2e/port_test.go:136]: I think it's important to have the latest tested still then.  Can we spin up new tests that skip remote with latest still?

Comment (edsantiago) [reply test/e2e/port_test.go:136]: Can you clarify whether you mean "it is important to test -l", or "it is important to test -l in this precise exact port test"? There are still a ton of `-l` tests.

**Comment by edsantiago** [comment]
Failure ([log](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5687800454447104/html/integration_test.log.html)) seems to be consistent, in [It] podman rm -v removes anonymous volume : 
```
podman [options]-remote --remote --url unix:/... volume list --quiet
b959d6c406db5ae242f34ded602dd6bb05ab631529c583b9b9a4122352939d1e    <<---- expected empty string
```
Recommendation: disable that test with a `SkipIfRemote()  // TODO: remote rm -v does not seem to work`

File: test/e2e/run_restart_test.go (hunk 5 above)
Comment (rhatdan) [reply test/e2e/run_restart_test.go:41]: Sure if we want to have one test to test --latest option, but I believe there is more value in testing these tests as remote.

File: test/e2e/run_networking_test.go (hunk 4 above)
Comment (rhatdan) [reply test/e2e/run_networking_test.go:269]: Yes, we do we care about testing the -l or --latest inspect flag when testing networking code.  We want to test the networking code for remote.

File: test/e2e/port_test.go (hunk 3 above)
Comment (rhatdan) [reply test/e2e/port_test.go:136]: We should probably have one test for each command on --latest but no more.  Just like we should test every flag.

Comment (rhatdan) [reply test/e2e/port_test.go:136]: I just checked there are three other tests in port_test.go that test `-l`, I would argue that we don't need this many, but these tests had -l in their test name, so I left them and skipped local.

Comment (TomSweeneyRedHat) [reply test/e2e/port_test.go:136]: @edsantiago  My main thing is we've coverage.  After hearing our potential low coverage numbers the other day, I hate to remove tests that potentially keep our numbers up a bit.  Not so much these particular tests, but I'm just overall leery about removing potential branches of testing without at least verifying we still have coverage.

**Comment by rhatdan** [comment]
Well we actually have very good testing in the integration and e2e,  Where we don't have good coverage is on the unit level.  Especially in libpod.  Basically we need to test all of the external functions,with unit tests.

**Pull request merged**